### PR TITLE
net: lib: azure_iot_hub: Fix user_name generation

### PR DESCRIPTION
**Pull request by alextsam**
Currently, the `user_name` is generated in runtime only when DPS is used
and in all other cases it is a build-time concatenation of Kconfig config 
parameters. However this creates an issue when using a device id that is not 
set in Kconfig but is set in runtime by the app. This patch adds runtime 
generation for the user_name for this case.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

### CODE REVIEW

File: subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ static azure_iot_hub_evt_handler_t evt_handler;
  * name + device ID length + lengths of ".azure-devices-provisioning.net/"
  * and "/?api-version=2018-06-30"
  */
-#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)
+#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS) || IS_ENABLED(CONFIG_AZURE_IOT_HUB_DEVICE_ID_APP)
```
Comment (coderbyheart) [inline subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c:78]: Documentation for this block should be updated.

File: subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c [hunk 2]
```diff
@@ -75,7 +75,7 @@ static azure_iot_hub_evt_handler_t evt_handler;
...
 #define USER_NAME_BUF_LEN	(CONFIG_AZURE_IOT_HUB_HOSTNAME_MAX_LEN + \
```
Comment (coderbyheart) [inline subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c:79]: I think it is OK that the buffer size includes `.azure-devices-provisioning.net/`, more correct would be to have two different `USER_NAME_BUF_LEN` definitions.

File: subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c [hunk 3]
```diff
@@ -879,6 +884,9 @@ static int client_broker_init(struct mqtt_client *const client, bool dps)
 		tls_cfg->hostname = dps_hostname_get();
 		client->user_name = user_name_get();
 	}
+#elif IS_ENABLED(CONFIG_AZURE_IOT_HUB_DEVICE_ID_APP)
+    tls_cfg->hostname = CONFIG_AZURE_IOT_HUB_HOSTNAME;
+	client->user_name = user_name_get();
```
Comment (coderbyheart) [inline subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c:889]:
```suggestion
	tls_cfg->hostname = CONFIG_AZURE_IOT_HUB_HOSTNAME;
	client->user_name = user_name_get();
```

**Review by coderbyheart — CHANGES_REQUESTED** [review]

**Review by jtguggedal — APPROVED** [review]
Thanks for fixing this issue!

**Comment by coderbyheart** [comment]
NordicBuilder wrote:
> At this time we are not accepting external contributions, but this may change soon.

Please excuse the wording. **We really do appreciate your contribution.** This message was put in place because we cannot run pull requests from untrusted sources automatically on our test infrastructure right now which limits our ability to quickly review a pull request. We are working on improving this situation.

File: subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c (hunk 2 above)
Comment (alextsam) [reply subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c:83]: Thanks for the comments! I just pushed an updated version with the define split up and updated comments

**Review by coderbyheart — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**